### DISCUSSION TITLE
Copilot 1.71.1.0 and Visual Studio 2022 version 17.4.33213 This extension is not installable on any currently installed products.

### DISCUSSION THREAD

**Discussion by doug-coates** [community/community#46881] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

Error log trying to install Copilot.... can you help?

2/9/2023 12:13:21 PM - Microsoft VSIX Installer
2/9/2023 12:13:21 PM - -------------------------------------------
2/9/2023 12:13:21 PM - vsixinstaller.exe version:
2/9/2023 12:13:21 PM - 17.4.2119
2/9/2023 12:13:21 PM - -------------------------------------------
2/9/2023 12:13:21 PM - Command line parameters:
2/9/2023 12:13:21 PM - C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service\vsixinstaller.exe,/appidinstallpath:C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\devenv.exe,/skuName:Enterprise,/skuVersion:17.4.33122.133,/appidname:Microsoft Visual Studio Enterprise 2022,/culture:en-US,/noep,C:\Users\wdcj\AppData\Local\Temp\VSIX25mz4o5a.vsix
2/9/2023 12:13:21 PM - -------------------------------------------
2/9/2023 12:13:21 PM - Microsoft VSIX Installer
2/9/2023 12:13:21 PM - -------------------------------------------
2/9/2023 12:13:22 PM - Initializing Install...
2/9/2023 12:13:22 PM - Extension Details...
2/9/2023 12:13:22 PM - 	Identifier         : GitHub.Copilot
2/9/2023 12:13:22 PM - 	Name               : GitHub Copilot
2/9/2023 12:13:22 PM - 	Author             : GitHub
2/9/2023 12:13:22 PM - 	Version            : 1.71.1.0
2/9/2023 12:13:22 PM - 	Description        : GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.
2/9/2023 12:13:22 PM - 	Locale             : en-US
2/9/2023 12:13:22 PM - 	MoreInfoURL        : 
2/9/2023 12:13:22 PM - 	InstalledByMSI     : False
2/9/2023 12:13:22 PM - 	SupportedFrameworkVersionRange : [0.0,2147483647.2147483647]
2/9/2023 12:13:22 PM - 
2/9/2023 12:13:23 PM - 	SignatureState     : ValidSignature
2/9/2023 12:13:23 PM - 	SignedBy           : GitHub, Inc.
2/9/2023 12:13:23 PM - 	Certificate Info   : 
2/9/2023 12:13:23 PM - 		-------------------------------------------------------
2/9/2023 12:13:23 PM - 		[Subject]       : CN="GitHub, Inc.", O="GitHub, Inc.", L=San Francisco, S=California, C=US
2/9/2023 12:13:23 PM - 		[Issuer]        : CN=DigiCert SHA2 Assured ID Code Signing CA, OU=www.digicert.com, O=DigiCert Inc, C=US
2/9/2023 12:13:23 PM - 		[Serial Number] : 09144B52C8330F257D2512DAAAA1E545
2/9/2023 12:13:23 PM - 		[Not Before]    : 2/4/2020 7:00:00 PM
2/9/2023 12:13:23 PM - 		[Not After]     : 2/9/2023 7:00:00 AM
2/9/2023 12:13:23 PM - 		[Thumbprint]    : 9C5A6D389E1454F2ED9EE9419CDF743689709F9C
2/9/2023 12:13:23 PM - 
2/9/2023 12:13:23 PM - 	Supported Products : 
2/9/2023 12:13:23 PM - 		Microsoft.VisualStudio.Community
2/9/2023 12:13:23 PM - 			Version : [17.4.33213,18.0)
2/9/2023 12:13:23 PM - 			ProductArchitecture : amd64
2/9/2023 12:13:23 PM - 
2/9/2023 12:13:23 PM - 	References         : 
2/9/2023 12:13:23 PM - 		-------------------------------------------------------
2/9/2023 12:13:23 PM - 		Identifier   : Microsoft.VisualStudio.IntelliCode.Internal.InlineSuggestions
2/9/2023 12:13:23 PM - 		Name         : Inline suggestions
2/9/2023 12:13:23 PM - 		Version      : [2.2.2404.12984,3.0)
2/9/2023 12:13:23 PM - 		MoreInfoURL  : 
2/9/2023 12:13:23 PM - 		Nested       : Yes
2/9/2023 12:13:23 PM - 
2/9/2023 12:13:23 PM - 	Prerequisites      : 
2/9/2023 12:13:23 PM - 		-------------------------------------------------------
2/9/2023 12:13:23 PM - 		Identifier   : Microsoft.VisualStudio.Component.IntelliCode
2/9/2023 12:13:23 PM - 		Name         : IntelliCode
2/9/2023 12:13:23 PM - 		Version      : [17.4.33213,18.0)
2/9/2023 12:13:23 PM - 
2/9/2023 12:13:23 PM - Signature Details...
2/9/2023 12:13:23 PM - 	Extension is signed with a valid signature.
2/9/2023 12:13:23 PM - 
2/9/2023 12:13:23 PM - VSIXInstaller.NoApplicableSKUsException: This extension is not installable on any currently installed products.
   at VSIXInstaller.ExtensionService.GetInstallableDataImpl(IInstallableExtension extension, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionService.GetInstallableData(String vsixPath, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionService.ComputeInstallableExtensions(IStateData stateData, Boolean isRepairSupported, Boolean isExtensionPack, IDictionary`2 skuDataMapping, IList`1 installableExtensionsPaths)
   at VSIXInstaller.App.Initialize(Boolean isRepairSupported)
   at VSIXInstaller.App.Initialize()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()

**Comment by tilmann1** [comment, marked as answer]
@doug-coates Please upgrade your Visual Studio installation to version 17.4.4 then you should be able to install the extension just fine, please see https://github.com/community/community/discussions/46819 for more details. Please let us know in case that doesn't solve your problem, thanks!

**Reply by reza-toorani** [reply]
I am getting the same error,
VS: Microsoft Visual Studio Professional 2022 (64-bit) - Current Version 17.5.1
OS: Windows 10 Enterprise

6/30/2023 10:35:45 AM - -------------------------------------------
6/30/2023 10:35:45 AM - Microsoft VSIX Installer
6/30/2023 10:35:45 AM - -------------------------------------------
6/30/2023 10:35:45 AM - Initializing Install...
6/30/2023 10:35:45 AM - Extension Details...
6/30/2023 10:35:45 AM - Identifier : GitHub.Copilot
6/30/2023 10:35:45 AM - Name : GitHub Copilot
6/30/2023 10:35:45 AM - Author : GitHub
6/30/2023 10:35:45 AM - Version : 1.93.0.0
6/30/2023 10:35:45 AM - Description : GitHub Copilot is an AI pair programmer that helps you write code faster and with less work. Requires Visual Studio 2022 17.4.4 or later.
6/30/2023 10:35:45 AM - Locale : en-US
6/30/2023 10:35:45 AM - MoreInfoURL :
6/30/2023 10:35:45 AM - InstalledByMSI : False
6/30/2023 10:35:45 AM - SupportedFrameworkVersionRange : [0.0,2147483647.2147483647]
6/30/2023 10:35:46 AM - SignatureState : ValidSignature
6/30/2023 10:35:46 AM - SignedBy : GitHub, Inc.
6/30/2023 10:35:46 AM - Certificate Info :
6/30/2023 10:35:46 AM - -------------------------------------------------------
6/30/2023 10:35:46 AM - [Subject] : CN="GitHub, Inc.", O="GitHub, Inc.", L=San Francisco, S=California, C=US
6/30/2023 10:35:46 AM - [Issuer] : CN=DigiCert Trusted G4 Code Signing RSA4096 SHA384 2021 CA1, O="DigiCert, Inc.", C=US
6/30/2023 10:35:46 AM - [Serial Number] : 0ED17E3276477EECDEDE0A65744ABC0E
6/30/2023 10:35:46 AM - [Not Before] : 2/10/2023 1:00:00 AM
6/30/2023 10:35:46 AM - [Not After] : 2/14/2024 12:59:59 AM
6/30/2023 10:35:46 AM - [Thumbprint] : 0C5C3D5588A23CC0B287C00262D03CC3F6F4A7FB
6/30/2023 10:35:46 AM - Supported Products :
6/30/2023 10:35:46 AM - Microsoft.VisualStudio.Community
6/30/2023 10:35:46 AM - Version : [17.5.33627,18.0)
6/30/2023 10:35:46 AM - ProductArchitecture : amd64
6/30/2023 10:35:46 AM - References :
6/30/2023 10:35:46 AM - Prerequisites :
6/30/2023 10:35:46 AM - -------------------------------------------------------
6/30/2023 10:35:46 AM - Identifier : Microsoft.VisualStudio.Component.IntelliCode
6/30/2023 10:35:46 AM - Name : IntelliCode
6/30/2023 10:35:46 AM - Version : [17.4.33213,18.0)
6/30/2023 10:35:46 AM - Signature Details...
6/30/2023 10:35:46 AM - Extension is signed with a valid signature.
6/30/2023 10:35:46 AM - VSIXInstaller.NoApplicableSKUsException: This extension is not installable on any currently installed products.
at VSIXInstaller.ExtensionService.GetInstallableDataImpl(IInstallableExtension extension, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable1& skuData) at VSIXInstaller.ExtensionService.GetInstallableData(String vsixPath, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable1& skuData)
at VSIXInstaller.ExtensionService.ComputeInstallableExtensions(IStateData stateData, Boolean isRepairSupported, Boolean isExtensionPack, IDictionary2 skuDataMapping, IList1 installableExtensionsPaths)
at VSIXInstaller.App.Initialize(Boolean isRepairSupported)
at VSIXInstaller.App.Initialize()
at System.Threading.Tasks.Task`1.InnerInvoke()
at System.Threading.Tasks.Task.Execute()

**Comment by AliceCanada** [comment]
I have installed VS professional Version 17.5.3, still get the same error, "This extension is not installable on any currently installed products". What can I do?

**Reply by ayim** [reply]
Hi Alice, this doesn't sound right to us. I haven't heard of this failure case before so super interested - drop me an email at ayim at github dot com so we can screenshare and investigate what's going on.

**Reply by AliceCanada** [reply]
Thanks! I send you email with details. Alice

**Reply by ayim** [reply]
I took a look at this....

1) The functionality in GitHub extension previously available for VS 2015-2019 is now included in the box for VS2022.
https://marketplace.visualstudio.com/items?itemName=GitHub.GitHubExtensionforVisualStudio

2) If you're looking for Copilot, this is the extension link:
https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

**Reply by AliceCanada** [reply]
I used the 1st link download the file (GitHub.VisualStudio-v2.11.106), when I install it, I got the error "This extension is not installable on any currently installed products." From the log file, it does found my VS2022 - "Found installed product - Visual Studio Professional 2022", but get VSIXInstaller.NoApplicableSKUsException: 

Does it mean from VS2022, Copilot can replace of the GitHub extension?

Thanks!

**Reply by aldoggr** [reply]
Hello,
I'm facing the same issue in a freshly installed VS2022 v17.5.4 (ARM64, Macbook Pro M2 and Parallels Win11). 
I suspect that ARM64 isn't supported?

**Reply by Moristar** [reply]
> Hello, I'm facing the same issue in a freshly installed VS2022 v17.5.4 (ARM64, Macbook Pro M2 and Parallels Win11). I suspect that ARM64 isn't supported?

I have exactly the same setup and also get the error. Looks like no luck for ARM. Although it works for pure Mac and VS Code.

**Reply by ObfuscationEqualsTrue** [reply]
i am getting the same error. this is the log:

29/06/2023 1:37:21 PM - Microsoft VSIX Installer
29/06/2023 1:37:21 PM - -------------------------------------------
29/06/2023 1:37:21 PM - vsixinstaller.exe version:
29/06/2023 1:37:21 PM - 17.5.4065+gbda5f544dc
29/06/2023 1:37:21 PM - -------------------------------------------
29/06/2023 1:37:21 PM - Command line parameters:
29/06/2023 1:37:21 PM - C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service\vsixinstaller.exe,/appidinstallpath:C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\devenv.exe,/skuName:Pro,/skuVersion:17.5.33414.496,/appidname:Microsoft Visual Studio Professional 2022,/culture:en-US,/noep,C:\Users\koan.stevenson\AppData\Local\Temp\VSIXgbpvd5ke.vsix
29/06/2023 1:37:21 PM - -------------------------------------------
29/06/2023 1:37:21 PM - Microsoft VSIX Installer
29/06/2023 1:37:21 PM - -------------------------------------------
29/06/2023 1:37:22 PM - Initializing Install...
29/06/2023 1:37:22 PM - Extension Details...
29/06/2023 1:37:22 PM - 	Identifier         : GitHub.Copilot
29/06/2023 1:37:22 PM - 	Name               : GitHub Copilot
29/06/2023 1:37:22 PM - 	Author             : GitHub
29/06/2023 1:37:22 PM - 	Version            : 1.93.0.0
29/06/2023 1:37:22 PM - 	Description        : GitHub Copilot is an AI pair programmer that helps you write code faster and with less work. Requires Visual Studio 2022 17.4.4 or later.
29/06/2023 1:37:22 PM - 	Locale             : en-US
29/06/2023 1:37:22 PM - 	MoreInfoURL        : 
29/06/2023 1:37:22 PM - 	InstalledByMSI     : False
29/06/2023 1:37:22 PM - 	SupportedFrameworkVersionRange : [0.0,2147483647.2147483647]
29/06/2023 1:37:23 PM - 	SignatureState     : ValidSignature
29/06/2023 1:37:23 PM - 	SignedBy           : GitHub, Inc.
29/06/2023 1:37:23 PM - 	Certificate Info   : 
29/06/2023 1:37:23 PM - 		-------------------------------------------------------
29/06/2023 1:37:23 PM - 		[Subject]       : CN="GitHub, Inc.", O="GitHub, Inc.", L=San Francisco, S=California, C=US
29/06/2023 1:37:23 PM - 		[Issuer]        : CN=DigiCert Trusted G4 Code Signing RSA4096 SHA384 2021 CA1, O="DigiCert, Inc.", C=US
29/06/2023 1:37:23 PM - 		[Serial Number] : 0ED17E3276477EECDEDE0A65744ABC0E
29/06/2023 1:37:23 PM - 		[Not Before]    : 10/02/2023 11:00:00 AM
29/06/2023 1:37:23 PM - 		[Not After]     : 14/02/2024 10:59:59 AM
29/06/2023 1:37:23 PM - 		[Thumbprint]    : 0C5C3D5588A23CC0B287C00262D03CC3F6F4A7FB
29/06/2023 1:37:23 PM - 	Supported Products : 
29/06/2023 1:37:23 PM - 		Microsoft.VisualStudio.Community
29/06/2023 1:37:23 PM - 			Version : [17.5.33627,18.0)
29/06/2023 1:37:23 PM - 			ProductArchitecture : amd64
29/06/2023 1:37:23 PM - 	References         : 
29/06/2023 1:37:23 PM - 	Prerequisites      : 
29/06/2023 1:37:23 PM - 		-------------------------------------------------------
29/06/2023 1:37:23 PM - 		Identifier   : Microsoft.VisualStudio.Component.IntelliCode
29/06/2023 1:37:23 PM - 		Name         : IntelliCode
29/06/2023 1:37:23 PM - 		Version      : [17.4.33213,18.0)
29/06/2023 1:37:23 PM - Signature Details...
29/06/2023 1:37:23 PM - 	Extension is signed with a valid signature.
29/06/2023 1:37:23 PM - VSIXInstaller.NoApplicableSKUsException: This extension is not installable on any currently installed products.
   at VSIXInstaller.ExtensionService.GetInstallableDataImpl(IInstallableExtension extension, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionService.GetInstallableData(String vsixPath, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionService.ComputeInstallableExtensions(IStateData stateData, Boolean isRepairSupported, Boolean isExtensionPack, IDictionary`2 skuDataMapping, IList`1 installableExtensionsPaths)
   at VSIXInstaller.App.Initialize(Boolean isRepairSupported)
   at VSIXInstaller.App.Initialize()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()

**Reply by reza-toorani** [reply]
I am getting the same error, 
VS:  Microsoft Visual Studio Professional 2022 (64-bit) - Current Version 17.5.1
OS: Windows 10 Enterprise


6/30/2023 10:35:45 AM - -------------------------------------------
6/30/2023 10:35:45 AM - Microsoft VSIX Installer
6/30/2023 10:35:45 AM - -------------------------------------------
6/30/2023 10:35:45 AM - Initializing Install...
6/30/2023 10:35:45 AM - Extension Details...
6/30/2023 10:35:45 AM - 	Identifier         : GitHub.Copilot
6/30/2023 10:35:45 AM - 	Name               : GitHub Copilot
6/30/2023 10:35:45 AM - 	Author             : GitHub
6/30/2023 10:35:45 AM - 	Version            : 1.93.0.0
6/30/2023 10:35:45 AM - 	Description        : GitHub Copilot is an AI pair programmer that helps you write code faster and with less work. Requires Visual Studio 2022 17.4.4 or later.
6/30/2023 10:35:45 AM - 	Locale             : en-US
6/30/2023 10:35:45 AM - 	MoreInfoURL        : 
6/30/2023 10:35:45 AM - 	InstalledByMSI     : False
6/30/2023 10:35:45 AM - 	SupportedFrameworkVersionRange : [0.0,2147483647.2147483647]
6/30/2023 10:35:46 AM - 	SignatureState     : ValidSignature
6/30/2023 10:35:46 AM - 	SignedBy           : GitHub, Inc.
6/30/2023 10:35:46 AM - 	Certificate Info   : 
6/30/2023 10:35:46 AM - 		-------------------------------------------------------
6/30/2023 10:35:46 AM - 		[Subject]       : CN="GitHub, Inc.", O="GitHub, Inc.", L=San Francisco, S=California, C=US
6/30/2023 10:35:46 AM - 		[Issuer]        : CN=DigiCert Trusted G4 Code Signing RSA4096 SHA384 2021 CA1, O="DigiCert, Inc.", C=US
6/30/2023 10:35:46 AM - 		[Serial Number] : 0ED17E3276477EECDEDE0A65744ABC0E
6/30/2023 10:35:46 AM - 		[Not Before]    : 2/10/2023 1:00:00 AM
6/30/2023 10:35:46 AM - 		[Not After]     : 2/14/2024 12:59:59 AM
6/30/2023 10:35:46 AM - 		[Thumbprint]    : 0C5C3D5588A23CC0B287C00262D03CC3F6F4A7FB
6/30/2023 10:35:46 AM - 	Supported Products : 
6/30/2023 10:35:46 AM - 		Microsoft.VisualStudio.Community
6/30/2023 10:35:46 AM - 			Version : [17.5.33627,18.0)
6/30/2023 10:35:46 AM - 			ProductArchitecture : amd64
6/30/2023 10:35:46 AM - 	References         : 
6/30/2023 10:35:46 AM - 	Prerequisites      : 
6/30/2023 10:35:46 AM - 		-------------------------------------------------------
6/30/2023 10:35:46 AM - 		Identifier   : Microsoft.VisualStudio.Component.IntelliCode
6/30/2023 10:35:46 AM - 		Name         : IntelliCode
6/30/2023 10:35:46 AM - 		Version      : [17.4.33213,18.0)
6/30/2023 10:35:46 AM - Signature Details...
6/30/2023 10:35:46 AM - 	Extension is signed with a valid signature.
6/30/2023 10:35:46 AM - VSIXInstaller.NoApplicableSKUsException: This extension is not installable on any currently installed products.
   at VSIXInstaller.ExtensionService.GetInstallableDataImpl(IInstallableExtension extension, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionService.GetInstallableData(String vsixPath, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionService.ComputeInstallableExtensions(IStateData stateData, Boolean isRepairSupported, Boolean isExtensionPack, IDictionary`2 skuDataMapping, IList`1 installableExtensionsPaths)
   at VSIXInstaller.App.Initialize(Boolean isRepairSupported)
   at VSIXInstaller.App.Initialize()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()

**Comment by Ichaangelina** [comment]
> @doug-coates Please upgrade your Visual Studio installation to version 17.4.4 then you should be able to install the extension just fine, please see https://github.com/community/community/discussions/46819 for more details. Please let us know in case that doesn't solve your problem, thanks!

**Comment by ObfuscationEqualsTrue** [comment]
for everyone who is already on version 17.4.4 and getting this error i suggest doing the following. upgrade to the very latest visual studio version (in my case it was up to 17,6). i suspect there is a version incompatibility with some of the copilot versions and some of the versions after 17.4.4

**Reply by tomsupis** [reply]
Yap, this helped me.

**Reply by mijumiju** [reply]
Thanks. It work for m e.
That 17.5.1 upgrade to 17.6.5 can help.

**Reply by james-chang-xrspace** [reply]
Thanks. It work for me.

**Reply by MrTorfick** [reply]
Thanks for your help.

**Reply by ayim** [reply]
Folks, we needed to bump this version to 17.5 - this is the new minimum version

**Reply by stanmerrill** [reply]
Thanks.  It worked for me!!   I really appreciate it.
